pkg/allocator/allocator_test.go

```diff
@@ -172,16 +173,18 @@ func (d *dummyBackend) ListIDs(ctx context.Context) (identityIDs []idpool.ID, er
 	d.mutex.Lock()
 	defer d.mutex.Unlock()
 
-	return maps.Keys(d.identities), nil
+	return slices.Collect(maps.Keys(d.identities)), nil
 }
 
 func (d *dummyBackend) ListAndWatch(ctx context.Context, handler CacheMutations, stopChan chan struct{}) {
 	d.mutex.Lock()
 	d.handler = handler
 
 	// Sort by ID to ensure consistent ordering
-	ids := maps.Keys(d.identities)
-	sort.Slice(ids, func(i, j int) bool { return ids[i] < ids[j] })
+	ids := slices.SortedStableFunc(maps.Keys(d.identities), func(id idpool.ID, id2 idpool.ID) int {
+		return int(id) - int(id2)
+	})
```

Comment:
Given that you are going to apply a few other suggestions as well, this could be simplified to:

```suggestion
	ids := slices.Sorted(maps.Keys(d.identities))
```

Reply:
Ah thanks, i thought i changed this part before 🫣